### PR TITLE
chore(cyclotron): modify janitor metric, expose depth limit

### PR DESCRIPTION
**Pull request by oliverb123**
I want to see which worker is lagging, and I want to be able to bump the depth limit with an environment variable

### CODE REVIEW

File: rust/cyclotron-core/src/ops/meta.rs [hunk 1]
```diff
@@ -6,18 +6,26 @@ use crate::{
...
 
-pub async fn count_total_waiting_jobs<'c, E>(executor: E) -> Result<u64, QueueError>
+pub async fn count_total_waiting_jobs<'c, E>(executor: E) -> Result<Vec<(u64, String)>, QueueError>
 where
     E: sqlx::Executor<'c, Database = sqlx::Postgres>,
 {
-    let res = sqlx::query!(
-        "SELECT COUNT(*) FROM cyclotron_jobs WHERE state = 'available' AND scheduled <= NOW()",
+    struct Count {
+        count: Option<i64>,
```
Comment (bretthoerner) [inline rust/cyclotron-core/src/ops/meta.rs:14]: Seems harmless, but is it really possible for a count/group by to come back with a null?

File: rust/cyclotron-node/src/manager.ts [hunk 2]
```diff
@@ -5,14 +5,15 @@ import { convertToInternalPoolConfig, serializeObject } from './helpers'
...
-    constructor(private config: { shards: CyclotronPoolConfig[] }) {
+    constructor(private config: { shards: CyclotronPoolConfig[], shardDepthLimit: number }) {
         this.config = config
     }
 
     async connect(): Promise<void> {
         return await cyclotron.maybeInitManager(
             JSON.stringify({
                 shards: this.config.shards.map((shard) => convertToInternalPoolConfig(shard)),
+                shard_depth_limit: this.config.shardDepthLimit,
```
Comment (bretthoerner) [inline rust/cyclotron-node/src/manager.ts:16]: Where does this get read on the Rust side?

**Review by bretthoerner — APPROVED** [review]

File: rust/cyclotron-node/src/manager.ts (hunk 2 above)
Comment (oliverb123) [reply rust/cyclotron-node/src/manager.ts:16]: It's an `Option<u64>` in the `ManagerConfig`, so serde should pick it up if it's set

File: rust/cyclotron-core/src/ops/meta.rs (hunk 1 above)
Comment (oliverb123) [reply rust/cyclotron-core/src/ops/meta.rs:14]: According to sqlx 🤷

File: rust/cyclotron-node/src/manager.ts (hunk 2 above)
Comment (bretthoerner) [reply rust/cyclotron-node/src/manager.ts:16]: Ah, gotcha, I figured it would be added as part of this PR, didn't see it was there already. 👍

**Pull request merged**